woocommerce/woocommerce
# Fix variable prod can be added to cart without variations #44868

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #44868  .

Added check for product type. Variable product should not be added to cart without selected variation
![eb8ff-clip-29kb](https://github.com/user-attachments/assets/c979a9b3-4fb7-4f6b-bc46-5cfe833f0a0b)

The reason for this behavior:
If $variation_id =0  then the variable product will go into  $product_data
`$product_data = wc_get_product( $variation_id ? $variation_id : $product_id );`

Since this is a variable product it does not pass this condition into the block where there is a check for an empty variation.
`if ( $product_data->is_type( 'variation' ) ) {`

Below in the code there are no conditions that will not allow this product to be added to the cart



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1) Take the id($variable_ID) of any variable product (important! this product must be created earlier, not in the same process where the test function is executed - add_to_cart())

2) Using an extension like [WP Console](https://wordpress.org/plugins/wp-console/), execute functions:
```
WC()->initialize_cart();
WC()->initialize_session();
WC()->cart->empty_cart(); // For a clean test
WC()->cart->add_to_cart($variable_ID,1,0);
```

3) Check the contents of the basket (the product should not be added):
`WC()->cart->get_cart_contents_count()`



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Added verification for variable product without a specified variation when programmatically calling add_to_cart()
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
